packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -0,0 +1,301 @@
+    });
+  }
+
+  void onChangeActiveOverlays() {
+    _checkOverlays(widget.game.overlays.value);
+    setState(() {
+      initialActiveOverlays = widget.game.overlays.value;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget internalGameWidget = _GameRenderObjectWidget(widget.game);
+
+    final hasBasicDetectors = hasBasicGestureDetectors(widget.game);
+    final hasAdvancedDetectors = hasAdvancedGesturesDetectors(widget.game);
+
+    assert(
+      !(hasBasicDetectors && hasAdvancedDetectors),
+      '''
+        WARNING: Both Advanced and Basic detectors detected.
+        Advanced detectors will override basic detectors and the later will not receive events
+      ''',
+    );
+
+    if (hasBasicDetectors) {
+      internalGameWidget = applyBasicGesturesDetectors(
+        widget.game,
+        internalGameWidget,
+      );
+    } else if (hasAdvancedDetectors) {
+      internalGameWidget = applyAdvancedGesturesDetectors(
+        widget.game,
+        internalGameWidget,
+      );
+    }
+
+    if (hasMouseDetectors(widget.game)) {
+      internalGameWidget = applyMouseDetectors(
+        widget.game,
+        internalGameWidget,
+      );
+    }
+
+    final stackedWidgets = [internalGameWidget];
+    _addBackground(context, stackedWidgets);
+    _addOverlays(context, stackedWidgets);
+
+    // We can use Directionality.maybeOf when that method lands on stable
+    final textDir = widget.textDirection ?? TextDirection.ltr;
+
+    return Directionality(
+      textDirection: textDir,
+      child: Container(
+        color: widget.game.backgroundColor(),
+        child: LayoutBuilder(
+          builder: (_, BoxConstraints constraints) {
+            widget.game.onResize(constraints.biggest.toVector2());
+            return FutureBuilder(
+              future: _gameLoaderFutureCache,
+              builder: (_, snapshot) {
+                if (snapshot.hasError) {
+                  if (widget.errorBuilder == null) {
+                    throw snapshot.error!;
+                  } else {
+                    return widget.errorBuilder!(context, snapshot.error!);
+                  }
+                }
+                if (snapshot.connectionState == ConnectionState.done) {
+                  return Stack(children: stackedWidgets);
+                }
+                return widget.loadingBuilder?.call(context) ?? Container();
+              },
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  List<Widget> _addBackground(BuildContext context, List<Widget> stackWidgets) {
+    if (widget.backgroundBuilder == null) {
+      return stackWidgets;
+    }
+    final backgroundContent = KeyedSubtree(
+      key: ValueKey(widget.game),
+      child: widget.backgroundBuilder!(context),
+    );
+    stackWidgets.insert(0, backgroundContent);
+    return stackWidgets;
+  }
+
+  List<Widget> _addOverlays(BuildContext context, List<Widget> stackWidgets) {
+    if (widget.overlayBuilderMap == null) {
```

Comment:
nit: save to a variable

Reply:
Not my change, I just moved this file, I would rather not mix my changes, which is already quite a lot, with unrelated things